README.md

```diff
@@ -394,6 +412,8 @@ becomes
     else
       ghc-options: -O0
 
+**Note:** Conditionals with `condition: false` are omitted from the generated
+`.cabal` file.
```

Comment:
@tfausak do you see any potential issues with this behavior.  One reason why I'm eager to do this is that it will make the fix for https://github.com/sol/hpack/issues/400 less convoluted.

Reply:
It also gives us cleaner `.cabal` files, which doesn't hurt.

Reply:
Seems safe to me!